java/statestore/src/test/java/sleeper/statestore/dynamodb/DynamoDBStateStoreIT.java

```diff
@@ -406,24 +409,24 @@ public void shouldGetFilesThatAreReadyForGC() throws StateStoreException {
             //      just been marked as ready for GC
             FileInfo fileInfo3 = FileInfo.wholeFile()
                     .filename("file3")
-                    .fileStatus(FileInfo.FileStatus.READY_FOR_GARBAGE_COLLECTION)
+                    .fileStatus(FileInfo.FileStatus.ACTIVE)
                     .partitionId("root")
                     .numberOfRecords(100L)
                     .lastStateStoreUpdateTime(file3Time)
                     .build();
             stateStore.fixTime(file3Time);
             stateStore.addFile(fileInfo3);
+            stateStore.atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFile(List.of(fileInfo3),
+                    fileInfoFactory.rootFile("compacted3", 100L));
 
             // When / Then 1
             stateStore.fixTime(file1GCTime);
             assertThat(stateStore.getReadyForGCFilenamesBefore(file2Time)).containsExactly("file1");
-            assertThat(stateStore.getReadyForGCFiles()).toIterable().containsExactly(fileInfo1);
 
             // When / Then 2
             stateStore.fixTime(file3GCTime);
             assertThat(stateStore.getReadyForGCFilenamesBefore(file3Time.plus(Duration.ofMinutes(1))))
                     .containsExactly("file1", "file3");
-            assertThat(stateStore.getReadyForGCFiles()).toIterable().containsExactly(fileInfo1, fileInfo3);
```

Comment:
The fixTime calls here aren't needed now that the time filter is passed directly in getReadyForGCFilenamesBefore.